app/src/main/java/org/mifos/selfserviceapp/presenters/LoanApplicationPresenter.java

```diff
@@ -63,13 +64,17 @@ public void onError(Throwable e) {
                     @Override
                     public void onNext(LoanTemplate loanTemplate) {
                         getMvpView().hideProgress();
-                        getMvpView().showLoanTemplate(loanTemplate);
+                        if (loanState == LoanState.CREATE) {
```

Comment:
@therajanmaurya In case of an update loan state after the loan template is loaded, current values are need to be set in the fields so I divided them into two functions one for new application where we only need to intialize template and second where we need to place the current values of the loan which needs to be updated after the template has been loaded.